system/HTTP/URI.php

```diff
@@ -177,11 +177,11 @@ public static function createURIString(
                 : ltrim($path, '/');
         }
 
-        if ($query) {
+        if ($query !== null && $query !== '') {
             $uri .= '?' . $query;
         }
 
-        if ($fragment) {
+        if ($fragment !== null && $fragment !== '') {
```

Comment:
cast to string

Reply:
Only `$fragment !== null` is sufficient.

Reply:
updated to ony check !==null